api/Worker.json

```diff
@@ -126,6 +135,59 @@
             "deprecated": false
           }
         },
+        "es2015_module": {
+          "__compat": {
+            "description": "Support for ECMAScript2015 module worker",
+            "support": {
+              "chrome": {
+                "version_added": "80"
+              },
+              "chrome_android": {
+                "version_added": "80"
+              },
+              "edge": {
+                "version_added": "80"
+              },
+              "firefox": {
+                "version_added": false
+              },
+              "firefox_android": {
+                "version_added": false
+              },
+              "ie": {
+                "version_added": false
+              },
+              "nodejs": {
+                "version_added": "12.17.0",
+                "partial_implementation": true,
+                "notes": "ES2015 is enabled based on file extension rather than type option."
```

Comment:
```suggestion
                "notes": "ECMAScript modules are enabled based on the file extension <code>.mjs</code> rather than the <code>type</code> option."
```

Reply:
Of course, to create more work for you, this also suggests we should add a sibling feature, `type` (description:  `Constructor <code>type</code> option`), to capture which engines support the option. I'd welcome this addition to the PR, but if you don't want to take on that work, let me know—we can open an issue for it.

Reply:
Note that Node.js supports ESM with `.js` extension when setting `"type": "module"` in the `package.json` file. That being said I think the current phrasing is good enough, as `.mjs` works no matter what `package.json` contains.

Reply:
Regarding the `Constructor <code>type</code> option`, I'm not ure that would add much more value since the `"classic"` value is the default, and the only alternative value `"module"` is covered by this PR. Do you think it's relevant to add a separate entry?

Reply:
Ah, I misunderstood your original note. I interpreted it to mean that Node doesn't support the constructor's `type` option and uses the file extension to determine the module type instead. The Node.js docs don't seem to show support for the `type` option, but I haven't tested it.

It sounds like the real behavior is rather more complex: it checks the `"type"` field in `package.json` or, if unset, the file extension. We can rewrite the note to reflect that (I'll open a new suggestion).

I'd say this reinforces the need for a type option feature: we have a case where ESM is supported but the mechanism for opting in to that support has different compatibility data from the support itself.